tangelo/toolboxes/post_processing/post_selection.py

```diff
@@ -135,3 +135,32 @@ def split_frequency_dict(frequencies, indices, desired_measurement=None):
         marginal_dict = post_select(frequencies, expected_outcomes)
 
     return midcirc_dict, marginal_dict
+
+
+def split_frequency_dict_for_last_n_digits(frequencies, n_to_keep):
+    """Marginalize the frequencies dictionary over the last_n_digits.
+    This splits the frequency dictionary into two frequency dictionaries
+    and aggregates the corresponding frequencies.
+    If desired_measurement is provided, the marginalized frequencies are
+    post-selected for that outcome on the mid-circuit measurements.
+
+    Args:
+        frequencies (dict): The input frequency dictionary
+        last_n_to_keep (int): The last n digits of the bitstring to store.
+
+    Returns:
+        dict: The frequencies for the last n digits
+        dict: The marginal frequencies for remaining indices (different bitstring lengths in general)
+    """
+
+    freq_dict = dict()
```

Comment:
just call call them freqs1 and freqs2. Your second name gives no information and we can clearly see they're dicts.

same for `n_to_keep`: why not just `n` ? your docstring makes it really clear